### PR TITLE
feat: CRUD budget

### PR DESCRIPTION
**Pull request by TrisTaam**
## What?
- Setup budget table for room database.
- Budget Screen: Show all budgets by 3 types of all categories.
- EditBudget Screen: Edit a budget's amount.
## Why?
- Budget is a important component for analystics screen.
## How?
- It includes budget model, budget repo in domain layer; budget entity, budget dao, budget repo impl in data layer using clean architecture.
## Screenshot
image
image

### CODE REVIEW

File: app/src/main/java/com/proptit/budgetbuddy/data/mapper/BudgetMapper.kt
Comment (thuanpham582002) [inline app/src/main/java/com/proptit/budgetbuddy/data/mapper/BudgetMapper.kt:1]: Use extension functions.

File: app/src/main/java/com/proptit/budgetbuddy/data/source/local/roomdb/BudgetBuddyDatabase.kt [hunk 1]
```diff
@@ -4,25 +4,53 @@ import androidx.room.Database
...
+    abstract val budgetDao: BudgetDao
+    abstract val categoryDao: CategoryDao
 
     companion object {
         const val DATABASE_NAME = "BudgetBuddyDatabase.db"
         val callback = object : Callback() {
             override fun onCreate(db: SupportSQLiteDatabase) {
                 super.onCreate(db)
                 db.execSQL("INSERT INTO users (name, date_of_birth, email, avatar_url, balance) VALUES ('Example', '1041354000', 'example@gmail.com', '', 0)")
+                val triggerInsertBudgetsAfterCategoryInsert = """
```
Comment (thuanpham582002) [inline app/src/main/java/com/proptit/budgetbuddy/data/source/local/roomdb/BudgetBuddyDatabase.kt:33]: Extract to a factory

File: app/src/main/java/com/proptit/budgetbuddy/data/mapper/BudgetMapper.kt
Comment (TrisTaam) [reply app/src/main/java/com/proptit/budgetbuddy/data/mapper/BudgetMapper.kt:1]: > Use extension functions.

I am used to using this mapper style in Java. In the future, i will take advantage of extension functions for mapper with Kotlin

File: app/src/main/java/com/proptit/budgetbuddy/presentation/util/ImageLoader.kt
Comment (thuanpham582002) [inline app/src/main/java/com/proptit/budgetbuddy/presentation/util/ImageLoader.kt:1]: util or Ext ?

Comment (TrisTaam) [reply app/src/main/java/com/proptit/budgetbuddy/presentation/util/ImageLoader.kt:1]: It's my fault. It must be a Ext file

File: app/src/main/java/com/proptit/budgetbuddy/data/source/local/roomdb/BudgetBuddyDatabase.kt (hunk 1 above)
Comment (TrisTaam) [reply app/src/main/java/com/proptit/budgetbuddy/data/source/local/roomdb/BudgetBuddyDatabase.kt]: I have done like Minh's feature.

**Pull request merged**